jvb/src/main/java/org/jitsi/videobridge/xmpp/MediaSourceFactory.java

```diff
@@ -491,24 +486,24 @@ public static String getOwner(SourcePacketExtension source)
             String ownerAttr = ssrcInfoPacketExtension.getAttributeAsString(SSRCInfoPacketExtension.OWNER_ATTR_NAME);
             if (ownerAttr != null)
             {
-                return getResourceFromJid(ownerAttr);
+                return getResourceOrDomainFromJid(ownerAttr);
             }
         }
         return null;
     }
 
     /**
-     * Given a string containing a JID, return its StringPrep'd Resource part,
-     * or {@link Resourcepart#EMPTY}, as a string.
+     * Given a string, if it is a valid JID with a Resource part return its StringPrep'd Resource part. Otherwise,
+     * return the string itself.
      *
-     * Helper for {@link #getOwner}
+     * Helper for {@link #getOwner}.
      */
-    private static String getResourceFromJid(String jid)
+    private static String getResourceOrDomainFromJid(String jid)
```

Comment:
I think this function name should change with your updated "whole-string" approach.